website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+            score=5,
+            justification="The answer is accurate and directly related to the question and context provided."
+        )]
+
+#  Define the custom metric
+faithfulness = mlflow.metrics.genai.make_genai_metric(
+    name="faithfulness",
+    definition="Assesses how well the answer relates to the question and provided context.",
+    grading_prompt="Score from 1-5, where 1 is not related at all and 5 is highly relevant and accurate.",
+    examples=examples)
+
+```
+
+Define out LLM function (in this case it can be any function that follows certain input /output formats that mlflow evalaute supports)
+
+```python
+# Using custom function
+def my_llm(inputs):
+    answers = []
+    system_prompt = "Please answer the following question in formal language based on the context provided."
+    for index, row in inputs.iterrows():
+        print('INPUTS:', row)
+        completion = openai.chat.completions.create(
+            model="gpt-3.5-turbo",
+            messages=[
+                {"role": "system", "content": system_prompt},
+                {"role": "user", "content": f"{row}"},
+            ],
+        )
+        answers.append(completion.choices[0].message.content)
+
+    return answers
+```
+
+Then our evaluation is similar to what we did before
+
+```python
+with mlflow.start_run() as run:
+    results = mlflow.evaluate(
+        my_llm,
+        eval_data,
+        model_type="text",
+        evaluators="default",
+        extra_metrics=[faithfulness],
+        evaluator_config={
+        "col_mapping": {
+            "inputs": "llm_inputs",
+           }}
+    )
+mlflow.end_run()
+```
+
+### GenAI Metrics
+
+As an alternative we can use mlflow built in metrics for genai using the same examples.
+
+MLflow offers a few pre-built metrics which uses LLM as the judge. Despite the difference under the hood, the usage is the same - put these metrics in the extra_metrics argument in mlflow.evaluate(). Here is the list of pre-built metrics:
+
+1. **[mlflow.metrics.genai.answer_similarity()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_similarity)**
+
+2. **[mlflow.metrics.genai.answer_correctness()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_correctness)**
+
+
+3. **[mlflow.metrics.genai.answer_relevance()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_relevance)**
+ 
+4. **[mlflow.metrics.genai.relevance()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.relevance)**
+
+5. **[mlflow.metrics.genai.faithfulness()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.faithfulness)**
```

Comment:
can just link to the docs here, or maybe cite a few examples and then point to the docs (MLflow offers a few pre-build LLM-as-judge mertrics including answer similarity, answer correctness, ...)

Reply:
I simplified this and left only the necessary info and redirected to docs for the rest